analyzer/evmnfts/httpmisc/http_misc.go

```diff
@@ -0,0 +1,49 @@
+package httpmisc
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const ClientTimeout = 30 * time.Second
+
+func GetWithContextWithClient(ctx context.Context, client *http.Client, url string) (*http.Response, error) {
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	return client.Do(req)
+}
+
+type NotOKError struct {
+	// Note: .error is the implementation of .Error, .Unwrap etc. It is not
+	// in the Unwrap chain. Use something like
+	// `NotOKError{fmt.Errorf("...: %w", err)}` to set up an
+	// instance with `err` in the Unwrap chain.
+	error
+}
+
+func (err NotOKError) Is(target error) bool {
+	if _, ok := target.(NotOKError); ok {
+		return true
+	}
+	return false
+}
+
+func ResponseOK(resp *http.Response) error {
+	if resp.StatusCode >= 500 {
+		if err := resp.Body.Close(); err != nil {
+			return fmt.Errorf("HTTP closing body: %w", err)
```

Comment:
Here and below: Let's include the HTTP status code in the error.

Reply:
ok

Reply:
included